frontend/src/components/common/CreateResourceButton.tsx

```diff
@@ -0,0 +1,210 @@
+            const apiVersion = newItems[index].apiVersion;
+            if (newItems.length === 1) {
+              msg = t('translation|Failed to create {{ kind }} {{ name }}.', { kind, name });
+            } else {
+              msg = t('translation|Failed to create {{ kind }} {{ name }} in {{ apiVersion }}.', {
+                kind,
+                name,
+                apiVersion,
+              });
+            }
+            setErrorMessage(msg);
+            setOpenDialog(true);
+            throw msg;
+          }
+        });
+      }
+    );
+  };
+
+  function handleSave(newItemDefs: KubeObjectInterface[]) {
+    let massagedNewItemDefs = newItemDefs;
+    const cancelUrl = location.pathname;
+
+    // check if all yaml objects are valid
+    for (let i = 0; i < massagedNewItemDefs.length; i++) {
+      if (massagedNewItemDefs[i].kind === 'List') {
+        // flatten this List kind with the items that it has which is a list of valid k8s resources
+        const deletedItem = massagedNewItemDefs.splice(i, 1);
+        massagedNewItemDefs = massagedNewItemDefs.concat(deletedItem[0].items);
+      }
+      if (!massagedNewItemDefs[i].metadata?.name) {
+        setErrorMessage(
+          t(`translation|Invalid: One or more of resources doesn't have a name property`)
+        );
+        return;
+      }
+      if (!massagedNewItemDefs[i].kind) {
+        setErrorMessage(t('translation|Invalid: Please set a kind to the resource'));
+        return;
+      }
+    }
+    // all resources name
+    const resourceNames = massagedNewItemDefs.map(newItemDef => newItemDef.metadata.name);
+    setOpenDialog(false);
+
+    const clusterName = getCluster() || '';
+
+    dispatch(
+      clusterAction(() => applyFunc(massagedNewItemDefs, clusterName), {
+        startMessage: t('translation|Applying {{ newItemName }}…', {
+          newItemName: resourceNames.join(','),
+        }),
+        cancelledMessage: t('translation|Cancelled applying {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        successMessage: t('translation|Applied {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        errorMessage: t('translation|Failed to apply {{ newItemName }}.', {
+          newItemName: resourceNames.join(','),
+        }),
+        cancelUrl,
+      })
+    );
+
+    dispatchCreateEvent({
+      status: EventStatus.CONFIRMED,
+    });
+  }
+
+  const defaultContentMap: { [key: string]: any } = {
+    'Config Map': BASE_EMPTY_CONFIG_MAP,
+    Secret: BASE_EMPTY_SECRET,
+    Lease: LEASE_DUMMY_DATA,
+    RuntimeClass: BASE_RC,
+  };
+
+  const getDefaultContent = () => defaultContentMap[resource] || '';
+
+  return (
+    <React.Fragment>
+      <ActionButton
+        color="primary"
+        description={t('translation|Create {{ resource }}', { resource })}
+        icon={'mdi:plus-circle'}
+        onClick={() => {
+          setOpenDialog(true);
+        }}
+      />
```

Comment:
This button needs to be within a AuthVisible component. Because the user may not have the permissions to create the resource.